src/shared/Core/Trace2.cs

```diff
@@ -0,0 +1,316 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+
+namespace GitCredentialManager;
+
+/// <summary>
+/// The different event types tracked in the TRACE2 tracing
+/// system.
+/// </summary>
+public enum Event
+{
+    Version = 0,
+    Start = 1,
+    Exit = 2
+}
+
+/// <summary>
+/// Represents the application's TRACE2 tracing system, which
+/// writes to a collector service that stores the trace information.
+/// </summary>
+public interface ITrace2 : IDisposable
+{
+    /// <summary>
+    /// Time at which application began executing.
+    /// </summary>
+    long ApplicationStartTime { get; }
+
+    /// <summary>
+    /// Add a listener that will receive/queue events and write them
+    /// to the appropriate place based on event target.
+    /// </summary>
+    /// <param name="listener">The listener to add.</param>
+    void AddListener(CollectorWriter listener);
+    
+    /// <summary>
+    /// Parse TRACE2 event target
+    /// </summary>
+    /// <param name="eventTarget">
+    /// The TRACE2 event target value. Supports the following formats:
+    /// * af_unix:<socket_type>:<absolute-pathname> -  writes to Unix Domain Socket
```

Comment:
sorry to nit, this should be "af_unix:[<socket_type>:]<absolute-pathname>"

that is, the socket type is optional.  however, i generally don't set it anymore. (I had a reason at the time.)
If the C# NamedPipeClientStream class only does one type (I forget which), then we might document that.
I'm not sure we need to explicitly throw an error if the eventTarget string contains the other type -- i'm sure
that the network stack will let us know....